src/test/java/me/qoomon/maven/gitversioning/GitVersioningExtensionIT.java

```diff
@@ -249,6 +250,74 @@ void commitVersioning_multiModuleProject() throws Exception {
+            pomModel.addModule("api");
+            pomModel.addModule("logic");
+
+            writeModel(projectDir.resolve("pom.xml").toFile(), pomModel);
+            writeExtensionsFile(projectDir);
+
+            VersionDescription branchVersionDescription = new VersionDescription();
+            branchVersionDescription.pattern = ".*";
+            branchVersionDescription.versionFormat = "${branch}-gitVersioning";
+            extensionConfig.branch.add(branchVersionDescription);
+            writeExtensionConfigFile(projectDir, extensionConfig);
+
+            Path apiProjectDir = Files.createDirectories(projectDir.resolve("api"));
+            Model apiPomModel = writeModel(apiProjectDir.resolve("pom.xml").toFile(), new Model() {{
+                setModelVersion(pomModel.getModelVersion());
+                setGroupId(pomModel.getGroupId());
+                setVersion(pomModel.getVersion());
+                setArtifactId("api");
+            }});
+
+            Path logicProjectDir = Files.createDirectories(projectDir.resolve("logic"));
+            Model logicPomModel = writeModel(logicProjectDir.resolve("pom.xml").toFile(), new Model() {{
+                setModelVersion(pomModel.getModelVersion());
+                setGroupId(pomModel.getGroupId());
+                setVersion(pomModel.getVersion());
+                setArtifactId("logic");
+            }});
+
+            // When
+            Verifier verifier = new Verifier(logicProjectDir.toFile().getAbsolutePath());
+            verifier.executeGoal("verify");
+            String log = getLog(verifier);
+
+            // Then
+            assertThat(log).doesNotContain("[ERROR]");
+            String expectedVersion = givenBranch.replace("/", "-") + "-gitVersioning";
+//        assertThat(log).contains("Building " + logicPomModel.getArtifactId() + " " + expectedVersion);
+            Model gitVersionedLogicPomModel = readModel(logicProjectDir.resolve("target/").resolve(GIT_VERSIONING_POM_NAME).toFile());
+            assertThat(gitVersionedLogicPomModel).satisfies(it -> assertSoftly(softly -> {
+                softly.assertThat(it.getModelVersion()).isEqualTo(logicPomModel.getModelVersion());
+                softly.assertThat(it.getGroupId()).isEqualTo(logicPomModel.getGroupId());
+                softly.assertThat(it.getArtifactId()).isEqualTo(logicPomModel.getArtifactId());
+                softly.assertThat(it.getVersion()).isEqualTo(expectedVersion);
+                softly.assertThat(it.getProperties()).doesNotContainKeys(
+                        "git.commit",
+                        "git.ref"
+                );
+            }));
+        } finally{
+            // remove .git dir forcibly, since sometimes the jupiter handler fails deleting git files on Windows
+            // see https://www.eclipse.org/lists/jgit-dev/msg01951.html
+            FileUtils.delete(projectDir.resolve(".git").toFile(),FileUtils.RECURSIVE | FileUtils.RETRY);
```

Comment:
My understanding is `projectDir` deletion is handled automatically because it is annotated with `@TempDir`

Reply:
You are right, @TempDir takes care of removing the temporary directory after the tests have been run. Unfortunately, at least on the Windows machine where I worked, the directory deletion fails to delete some of the files created by JGit, and I think this is due to the issue I linked in the comment. However, I 've been able to run the test without problems.
I had no time to polish the code and try to run it in a different environment, so feel free to remove that hacky line.